supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalJobExecutionFSM.scala

```
@@ -0,0 +1,248 @@
+  case class Run(executePromise: Promise[BackendJobExecutionResponse])
+  case class Abort(abortPromise: Promise[JobAbortResponse])
+  sealed trait JobState
+  case object Idle extends JobState
+  case object Running extends JobState
+  case object Aborting extends JobState
+  protected case class StateData(process: Option[Process])
+
+  val ProcessKilledCode = 143
+  val logger = LoggerFactory.getLogger("LocalBackend")
+  // TODO Support GCS ?
+  val fileSystems = List(FileSystems.getDefault)
+
+  private def splitFqn(fullyQualifiedName: FullyQualifiedName): (String, String) = {
+    val lastIndex = fullyQualifiedName.lastIndexOf(".")
+    (fullyQualifiedName.substring(0, lastIndex), fullyQualifiedName.substring(lastIndex + 1))
+  }
+
+  case class Command(argv: Seq[String]) {
+    override def toString = argv.map(s => "\"" + s + "\"").mkString(" ")
+  }
+
+  def props(jobDescriptor: BackendJobDescriptor, backendConfiguration: Config) = Props(new LocalJobExecutionFSM(jobDescriptor, backendConfiguration))
+}
+
+class LocalJobExecutionFSM(jobDescriptor: BackendJobDescriptor,
+                           backendConfiguration: Config) extends LoggingFSM[JobState, StateData] with SharedFileSystem {
+  import LocalJobExecutionActor._
+  import better.files._
+  import cromwell.core.PathFactory._
+  import scala.sys.process._
+
+  startWith(Idle, StateData(None))
+
+  when(Idle) {
+    case Event(Run(executePromise), data) => execute(executePromise, data)
+    case Event(Abort(abortPromise), _) =>
+      abortPromise success BackendJobExecutionAbortFailedResponse(jobDescriptor.key, new RuntimeException("Received Abort message but job was not running."))
+      context.stop(self)
+      stay()
+  }
+
+  when(Running) {
+    case Event(Abort(abortPromise), data) => abort(abortPromise, data)
+    case Event(Run(_), data) =>
+      logger.warn("Received Run message before but job is already running. Ignoring.")
+      stay()
+  }
+
+  whenUnhandled { NullFunction }
```

silently ignoring unrecognized messages doesn't seem like the best policy... 😄